app/helpers/pricing_helper.rb

```diff
@@ -1,97 +1,144 @@
 # frozen_string_literal: true
 
 module PricingHelper
+
   def pricing_html(pricing_type, product, account, currency, unit, display_for = nil, options = {})
-    permission = ("view_retail_pricing_" + currency.to_s).to_sym if ([pricing_type] & [:base_retail, :your_retail, :basic_pricing, :outlet_pricing]).present?
-    permission = ("view_net_pricing_" + currency.to_s).to_sym if ([pricing_type] & [:piece_base, :your_piece, :base_net, :your_net, :outlet_pricing]).present?
-
-    if current_user.can_view_pricing? && current_user.has_role?(permission)
-      price = case pricing_type
-              when :your_retail
-                # Single unit product price with your account discount * markup
-                product.your_price_retail(account, currency, unit)
-              when :base_retail
-                # Single unit product price net * 2 || Piece net price from product info * 2
-                product.base_price__retail(currency, unit, :single, account)
-              when :your_net, :your_piece
-                # Single unit product price with your account discount
-                current_account.showroom? && !current_user.has_role?(:view_your_net_price) ? nil : product.your_price_net(account, currency, unit)
-              when :base_net, :piece_base
-                # Single unit product price || Piece price from product info
-                product.base_price__net(currency, unit, :single, account)
-              when :basic_pricing, :outlet_pricing
-                # Product detail page pricing. Product price * 2 (No markup)
-                product.retail_price(account: account, currency: currency, unit: unit, type: :single)
-              end
-    end
-
-    return unless price
-
-    human_price = humanized_pricing(product, price, currency, display_for, options)
-
-    content_tag(:div, class: "product-price") do
-      if clarencehouse?
-        format_price(human_price)
-      elsif pricing_type == :outlet_pricing
-        outlet_price = product.outlet_price(account, currency, unit)
-        markup = Markup.find_for_product(account: account, product: product)
-        outlet_price = markup.apply_to(outlet_price)
-        outlet_price = format_price(outlet_price)
-        html = content_tag(:span, outlet_price, class: "#{currency}-price outlet-price line-through discontinued", data: { price: price })
-        html += content_tag(:span, human_price, class: "#{currency}-price", data: { price: price })
-        html
-      else
-        content_tag(:span, human_price, class: "#{currency}-price", data: { price: price })
+    Pricing.new(pricing_type, product, account, currency, unit, display_for = nil, options = {}, current_user).return_formatted_price
+  end
+
```

Comment:
The diff for this file isn't a great reflection of the changes because so much moved around. It's probably best viewed as the whole file.